sdk/identity/identity/src/credentials/authorizationCodeCredential.ts

```diff
@@ -130,12 +130,16 @@ export class AuthorizationCodeCredential implements TokenCredential {
    *                TokenCredential implementation might make.
    */
   async getToken(scopes: string | string[], options: GetTokenOptions = {}): Promise<AccessToken> {
-    return trace(`${this.constructor.name}.getToken`, options, async (newOptions) => {
-      const arrayScopes = Array.isArray(scopes) ? scopes : [scopes];
-      return this.msalFlow.getToken(arrayScopes, {
-        ...newOptions,
-        disableAutomaticAuthentication: this.disableAutomaticAuthentication,
-      });
-    });
+    return tracingClient.withSpan(
+      `${this.constructor.name}.getToken`,
```

Comment:
I went back and forth on doing this `this.constructor.name` interpolation - I just wasn't sure what that might look like after code has been minified / uglified. I honestly don't know, but it would be worth experimenting...

Reply:
I didn’t understand 🤔

Reply:
Imagine that you're using this code in a browser, and in production you run your bundle through a minifier / uglifier. 

What will the name of the class be? It might just be `r` or some other letter due to minification. Taking terser for example:

I created this file:

```js
class Foo {
  bar() {
    return this.constructor.name;
  }
}

console.log(new Foo().bar());
```

```
PS D:\workspace\onetime> npx terser test.js --compress --mangle toplevel
> class r{bar(){return this.constructor.name}}console.log((new r).bar());
```

So then your span ends up being `r.getToken`. At least that's what I'm worried about - maybe its not valid though but hopefully that explains my comment a little better?

I don't know, is that a valid concern? I'm honestly not sure...

Reply:
ohhh wow!! I see!

I guess minifiers in general allow different degress of minimization 🤔 so devs (since they have to bundle this themselves) could probably also make the minification less strict 🤔

Reply:
Yeah, I think that's totally fair! Like I said I was going back and forth myself so I am happy to defer to you on it -just wanted to share my thinking here 😄